pkg/operator/controller.go

```diff
@@ -601,9 +601,13 @@ func (c *Controller) syncProfile(tuned *tunedv1.Tuned, nodeName string) error {
 		// The Tuned daemon profile "tunedProfileName" for nodeName matched with MachineConfig
 		// labels set for additional machine configuration.  Sync the operator-created
 		// MachineConfig for MachineConfigPools 'pools'.
-		err := c.syncMachineConfig(getMachineConfigNameForPools(pools), mcLabels, profile.Status.Bootcmdline, profile.Status.Stalld)
-		if err != nil {
-			return fmt.Errorf("failed to update Profile %s: %v", profile.Name, err)
+		if profile.Status.TunedProfile == tunedProfileName && profileApplied(profile) {
```

Comment:
If the condition is false, should we requeue / return err? Or will the profile status be updated by the operand once the profile is applied, triggering operator reconciliation?

Reply:
The latter. the profile status be updated by the operand once the profile is applied, triggering operator reconciliation.